src/lib/OpenCvProvider.js

```diff
@@ -6,53 +6,39 @@ const { Consumer: OpenCvConsumer, Provider } = OpenCvContext
 
 export { OpenCvConsumer, OpenCvContext }
 
-const scriptId = 'opencv-react'
-const moduleConfig = {
-  wasmBinaryFile: 'opencv_js.wasm',
-  usingWasm: true
-}
-
-export const OpenCvProvider = ({ openCvPath, children, onLoad }) => {
-  const [loaded, setLoaded] = React.useState(false)
-
-  const handleOnLoad = React.useCallback(() => {
-    if (onLoad) {
-      onLoad(window.cv)
-    }
-    setLoaded(true)
-  }, [])
+export const OpenCvProvider = ({ openCvVersion = '4.5.5', openCvPath = '', children }) => {
+  const [cvInstance, setCvInstance] = React.useState({ loaded: false, cv: undefined });
 
   React.useEffect(() => {
+    const scriptId = 'opencv-react'
     if (document.getElementById(scriptId) || window.cv) {
-      setLoaded(true)
       return
     }
 
+    window.Module = window.Module || {
+      wasmBinaryFile: 'opencv_js.wasm',
+      usingWasm: true,
+      onRuntimeInitialized: () => {
+        console.log('opencv loaded', window.cv)
+        setCvInstance({ loaded: true, cv: window.cv });
+      }
+    }
+
```

Comment:
shouldn't we reinstatiate it if opencvpath or opencvversion changes?
And why are you worried about checking window.Module first if this effect only reruns when one of the two changes?

Reply:
about reinstantiating: in your original code you exit the `useEffect` if there's already a `window.cv` instance available https://github.com/giacomocerquone/opencv-react/blob/1d15d74346cb2cd583852a9ff9fe895d55cf9a44/src/lib/OpenCvProvider.js#L26 so none of this happens when opencvpath or opencvversion changes, and i'm not sure if loading multiple cv instances is something we want to allow.

Reply:
about putting the instance into state, i did some external tests and it doesn't seem to be encumbering the state. if you don't put it into the state, the users of `useOpenCv` actually have to take the instance themselves from `window.cv` which will lead to non-encapsulation of the loading process, and some more checks on their behalf on each use.
and i think it is nicer this way...